framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,236 +86,158 @@ private static Pattern anchored(String regex) {
+            memberSelected = expr.getName().asString();
             receiver = parseHelper(receiverString, context, path);
         }
 
-        if (memberSelected.equals("class")) {
-            if (receiver instanceof FlowExpressions.ClassName && !context.parsingMember) {
-                return receiver;
-            } else {
-                throw constructParserException(s, "class is not a legal identifier");
-            }
-        }
-
         // Parse the rest, with a new receiver.
         FlowExpressionContext newContext = context.copyChangeToParsingMemberOfReceiver(receiver);
         return parseHelper(memberSelected, newContext, path);
     }
 
-    // ########
-
-    private static boolean isNullLiteral(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
-        }
-        return s.equals("null");
-    }
-
-    private static Receiver parseNullLiteral(Types types) {
-        return new ValueLiteral(types.getNullType(), (Object) null);
-    }
-
-    private static boolean isIntLiteral(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
-        }
-        Matcher intMatcher = INT_PATTERN.matcher(s);
-        return intMatcher.matches();
-    }
-
-    private static Receiver parseIntLiteral(String s, Types types) {
-        int val = Integer.parseInt(s);
-        return new ValueLiteral(types.getPrimitiveType(TypeKind.INT), val);
-    }
-
-    private static boolean isLongLiteral(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
-        }
-        Matcher longMatcher = LONG_PATTERN.matcher(s);
-        return longMatcher.matches();
-    }
-
-    private static Receiver parseLongLiteral(String s, Types types) {
-        // Remove L or l at the end of a long literal
-        s = s.substring(0, s.length() - 1);
-        long val = Long.parseLong(s);
-        return new ValueLiteral(types.getPrimitiveType(TypeKind.LONG), val);
-    }
+    private static Receiver parseClassExpression(
+            ClassExpr expr, ProcessingEnvironment env, FlowExpressionContext context, TreePath path)
+            throws FlowExpressionParseException {
+        Resolver resolver = new Resolver(env);
 
-    private static boolean isFloatLiteral(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
-        }
-        Matcher floatMatcher = FLOAT_PATTERN.matcher(s);
-        return floatMatcher.matches();
-    }
+        Pair<ClassName, String> classAndRemainingString =
+                matchPackageAndClassNameWithinExpression(expr.toString(), resolver, path);
 
-    /** Return true iff s is a string literal. */
-    private static boolean isStringLiteral(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
+        if (classAndRemainingString != null) {
```

Comment:
Similar question - could we not just avoid this call entirely and use a call to `parseHelper` with `expr.getTypeAsString()` as the first argument?